kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -71,6 +74,8 @@ public S3OutputStream(String key, S3SinkConnectorConfig conf, AmazonS3 s3) {
     this.bucket = conf.getBucketName();
     this.key = key;
     this.ssea = conf.getSsea();
+    this.sseKey = conf.getSseKey();
+    this.sseCustomerKey = null;
```

Comment:
We have all we need to know in the constructor to create the key object or not. 
Thus, I'd suggest making this variable `final` and setting it here as: 
`this.sseCustomerKey = (condition below) ? new key : null;`